### DISCUSSION TITLE
URL with token and username parameter redirecting back to login page

### DISCUSSION THREAD

**Discussion by opbissa** [pimcore/pimcore#16409] [Q&A / Support]
I am implementing SAML login using Office 365, everything is working fine and once the user is redirected back after successful authentication it generates the login URL with the following code

        $token = Authentication::generateToken($user->getName()); 
        $loginUrl = $this->generateUrl('pimcore_admin_login_check', [
                    'username' => $username,
                    'token' => $token,
        ], UrlGeneratorInterface::ABSOLUTE_URL);

        return $this->redirect($loginUrl);

but the user is redirected back to the login page, I tried to copy the same URL and pasted it in a new tab and it worked fine.

**Comment by patriciomg** [comment]
Hi, 
newbie here, did you solve it? 
I am just wondering, why are you generating an URL as if you were going to share it (in an email maybe). Isn't it that you should use **redirectToRoute('route-name')** and then in a SamlLoginCheck Controller Class check if username and token are fine, look for the user, then do $this->authenticateUser($user) and finally call the admin route with redirectToRoute ?   I shortly finished the tutorials but well, that is not yet enough to do more funny tasks :)
